### PR TITLE
invoices: properly set Preimage field for success resolution

### PR DESCRIPTION
**Pull request by Crypt-iQ**
Fixes https://github.com/lightningnetwork/lnd/issues/6436

Instead of setting the preimage to `inv.Terms.PaymentPreimage`, it is now set to `htlc.AMP.Preimage` if the former is nil. Also removed a redundant comparison since the `HTLCSet` logic already checks it.

### CODE REVIEW

File: invoices/update.go [hunk 1]
```diff
@@ -196,13 +204,6 @@ func updateMpp(ctx *invoiceUpdateCtx,
 	// Check whether total amt matches other htlcs in the set.
 	var newSetTotal lnwire.MilliSatoshi
 	for _, htlc := range htlcSet {
-		// Only consider accepted mpp htlcs. It is possible that there
```
Comment (Roasbeef) [inline invoices/update.go:199]: How is this related to the greater fix? 

Do we have a regression test we can commit here, perhaps derived of what you used to originally reproduce the issue?

Comment (Crypt-iQ) [reply invoices/update.go:199]: Maybe if a hodl flag was added

**Review by Roasbeef — APPROVED** [review]
LGTM ✨
